cmd/dagger/logger/tty.go

```diff
@@ -86,11 +86,7 @@ func (l *Logs) Add(event Event) error {
 		if err != nil {
 			return err
 		}
-
-		if group.FinalState.CanTransition(t) {
```

Comment:
This was so that we "only" go forward in state:

- `actions.test_a` finishes with ERROR
- `actions.test_b` finishes with COMPLETED

Without this, FinalState will be `COMPLETED` whereas this guarantees we keep the most progressed state.

COMPUTING > COMPLETED > CANCELED > ERROR

(so that if `_b` fails with ERROR and `_a` is cancelled, FinalState is still ERROR)

Reply:
Perhaps instead of keeping `CanTransition` we could just have a `if t > group.FinalState` or a bespoke `if` condition (I feel uneasy relying on the order of the enum for this -- if something adds a `SKIPPED` state or something this will break)

Reply:
Good catch. I removed this this condition https://github.com/dagger/dagger/pull/2344/files#diff-ce49242492fa98816ae76d8b1ba6311eae1bd3ed70fc2731a85c016c8d071765L53 where s == StateComputing was causing some issues when tasks moved from `completed` to `fail` for hidden actions.  This condition was added to handle the Dockerfile scenario, now that we don't have it anymore I can bring it back.